### ISSUE TITLE
with_scope Tag does not work in Wagon

### ISSUE THREAD

**Issue by westonganger** [locomotivecms/engine#1330]
It appears that the `with_scope` tag doesnt actually filter the array items while developing in wagon. 

For example the following will actually loop through all blocks regardless of the specified type

```
{% with_scope type: 'pricing_plan' %}
{% for block in section.blocks %}
  {{ block.type }}
{% endfor %}
{% endwith %}
```

Two suggestions:

A. Within the code add a fallback that utilizes a regular `array.select` when testing in wagon
B. Add documentation that states what the expected behaviour is

**Comment by DonKoko** [comment]
That is indeed the case, however I am not sure if its a wagon issue. From what I tested before, i believe that `{% with_scope %}` doesn't work with section blocks.

**Comment by westonganger** [comment]
@DonKoko yeah I think you are right about that because it was working fine for other stuff.

**Comment by did** [comment]
@westonganger @DonKoko, I do prefer the following syntax:

```liquid
{% for block in section.blocks %}
  {% if block.type == 'pricing_plan' %}
  {% endif %}
{% endfor %}
```

or even, 

```liquid
{% for block in section.blocks %}
  {% case shipping_method.title %}
  {% when 'pricing_plan' %}
  ...
  {% when 'another_bock_type' %}
  ...
  {% else %}
  {% endcase %}
{% endblock %}
```

What do you think?

**Comment by westonganger** [comment]
Yes that actually is prefferred usage however you never know what someone wants to achieve.

The main point is that the `with_scope` method must behave correctly for all collection types.
